cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltActivity.java

```diff
@@ -350,6 +362,54 @@ protected void onCreate(Bundle savedInstanceState) {
     videoSurfaceView = new VideoSurfaceView(this);
     addContentView(
         videoSurfaceView, new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
+
+    initializeJavaBridge();
+  }
+
+  /**
+   * Initializes the Java Bridge to allow communication between Java and JavaScript.
+   * This method injects Java objects into the WebView and loads corresponding JavaScript code.
+   */
+  private void initializeJavaBridge() {
+
+    WebContents webContents = getActiveWebContents();
+    if (webContents == null) {
+        // WebContents not initialized yet, post a delayed runnable to check again
+        new Handler(Looper.getMainLooper()).postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                initializeJavaBridge(); // Recursive call to check again
+            }
+        }, JAVA_BRIDGE_INITIALIZATION_DELAY_MILLI_SECONDS);
+        return;
+    }
+
+    // --- Initialize the Java Bridge ---
+
+    // 1. Gather all Java objects that need to be exposed to JavaScript.
+    javaScriptAndroidObjectList.add(new AmatiDeviceInspector(this));
+
+    // 2. Use JavascriptInjector to inject Java objects into the WebContents.
+    //    This makes the annotated methods in these objects accessible from JavaScript.
+    JavascriptInjector javascriptInjector = JavascriptInjector.fromWebContents(webContents, false);
+    if (javascriptInjector == null) {
```

Comment:
This looks quite fragile. What are reasons for this JavascriptInjdector to fail? Are those failure transient? Can we retry? Maybe add a TODO comment for now.

Reply:
The code will hit https://source.chromium.org/chromium/chromium/src/+/main:content/public/android/java/src/org/chromium/content/browser/JavascriptInjectorImpl.java;l=48;drc=0afc9ac9afcaab79fc54299039f4d27abf3a086d and https://source.chromium.org/chromium/chromium/src/+/main:content/public/android/java/src/org/chromium/content/browser/webcontents/WebContentsImpl.java;drc=0afc9ac9afcaab79fc54299039f4d27abf3a086d;l=1042

If we do not wait JAVA_BRIDGE_INITIALIZATION_DELAY_MILLI_SECONDS, webContents can be null, in my local test, that will cause JavascriptInjector.fromWebContents() will run into an exception.

In the case of webContents is not null, I never see the javascriptInjector is empty, we can remove the if check here.

Reply:
Don't create JavascriptInjector from null WebContets. Make sure you have the WebContents instance initialized first. Also check if you can leverage a callback/notification about when it is created.

Reply:
I tried to move it into CobaltContentBrowserClient::OnWebContentsCreated(), draft code is #4371.
We called the JavaBridge APIs same as this PR, in the implementation of #4371, after `webContents.evaluateJavaScript(jsCode, null);` is called, the Android java method annotated with @CobaltJavaScriptInterface is not triggered. But if we keep it here, everything works.